common/trinary/trit_tryte.c

```diff
@@ -68,34 +67,33 @@ uint8_t set_trit_at(tryte_t *const trytes, size_t const length, size_t index,
 
 void trits_to_trytes(trit_t const *const trits, tryte_t *const trytes,
                      size_t const length) {
-  int j = 0,
-      end = length < NUMBER_OF_TRITS_IN_A_TRYTE ? length
-                                                : NUMBER_OF_TRITS_IN_A_TRYTE,
-      i = end;
-  for (; i-- > 0;) {
-    j *= 3;
-    j += trits[i];
-  }
-  // int j = trits[0] + trits[1] * 3 + trits[2] * 9;
-  if (j < 0) {
-    j += TRYTE_SPACE;
-  }
-  trytes[0] = TRYTE_ALPHABET[(size_t)j];
-  if (length <= NUMBER_OF_TRITS_IN_A_TRYTE) {
-    return;
+  int k = 0;
```

Comment:
Cool, this function now supports trit arrays of more than just 3 trits, but what did the performance gain in this func?

Reply:
It already supported trit arrays of more than just 3 trits but via recursion, switching from recursion to iteration had a some performance gain

Reply:
just a nit, but this should probably've been an `int8_t`

Reply:
I actually tried that but performances were worse. I guess it's because CPUs are more efficient working on their word size ?